test/algorithm.cpp

```diff
@@ -87,4 +87,68 @@ TEST(algorithm_move_scalar) {
   ASSERT(a == b);
 }
 
+namespace {
+
+int pattern_0d() { return 8; }
+
+int pattern_1d(std::tuple<index_t> x) { return 2 * std::get<0>(x); }
```

Comment:
Can you use the existing `pattern`/`check_pattern` helpers? https://github.com/dsharlet/array/blob/master/test/test.h#L143

Reply:
Let me try this first before I submit.